File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        dict = {"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
        total = dict[s[len(s)-1]]
        for ch in range(0,len(s)-1):
            sign = -1 if dict[s[ch]]<dict[s[ch+1]] else 1
            total +=sign*dict[s[ch]]
        return total

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        String str = "IVXLCDM";
        HashMap<Character,Integer> map = new HashMap<>();

        int [] numericals = {1,5,10,50,100,500,1000};

        for(int i =0;i<str.length();i++){
            map.put(str.charAt(i),numericals[i]);
        }

        int total = map.get(s.charAt(s.length()-1));
        for(int i =0;i<s.length()-1;i++){
            int sign = map.get(s.charAt(i))<map.get(s.charAt(i+1)) ? -1:1;
            total += sign*map.get(s.charAt(i));
        }
        return total;
    }
}